wallet-standard/cryptid-wallet/src/wallet.ts

```diff
@@ -0,0 +1,272 @@
+        }
+
+        this.#connected();
+
+        return { accounts: this.accounts };
+    };
+
+    #disconnect: DisconnectMethod = async () => {
+        await this.#cryptidWallet.disconnect();
+    };
+
+    #signAndSendTransaction: SolanaSignAndSendTransactionMethod = async (...inputs) => {
+        if (!this.#account) throw new Error('not connected');
+
+        const outputs: SolanaSignAndSendTransactionOutput[] = [];
+
+        if (inputs.length === 1) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const { transaction, account, chain, options } = inputs[0]!;
+            const { minContextSlot, preflightCommitment, skipPreflight, maxRetries } = options || {};
+            if (account !== this.#account) throw new Error('invalid account');
+            if (!isSolanaChain(chain)) throw new Error('invalid chain');
+
+            const { signature } = await this.#cryptidWallet.signAndSendTransaction(
+                VersionedTransaction.deserialize(transaction),
+                {
+                    preflightCommitment,
+                    minContextSlot,
+                    maxRetries,
+                    skipPreflight,
+                }
+            );
+
+            outputs.push({ signature: bs58.decode(signature) });
+        } else if (inputs.length > 1) {
+            for (const input of inputs) {
+                outputs.push(...(await this.#signAndSendTransaction(input)));
+            }
+        }
+
+        return outputs;
+    };
+
+    #signTransaction: SolanaSignTransactionMethod = async (...inputs) => {
+        if (!this.#account) throw new Error('not connected');
+
+        const outputs: SolanaSignTransactionOutput[] = [];
+
+        if (inputs.length === 1) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const { transaction, account, chain } = inputs[0]!;
+            if (account !== this.#account) throw new Error('invalid account');
+            if (chain && !isSolanaChain(chain)) throw new Error('invalid chain');
+
+            const signedTransaction = await this.#cryptidWallet.signTransaction(
+                VersionedTransaction.deserialize(transaction)
+            );
+
+            outputs.push({ signedTransaction: signedTransaction.serialize() });
+        } else if (inputs.length > 1) {
+            let chain: SolanaChain | undefined = undefined;
+            for (const input of inputs) {
+                if (input.account !== this.#account) throw new Error('invalid account');
+                if (input.chain) {
+                    if (!isSolanaChain(input.chain)) throw new Error('invalid chain');
+                    if (chain) {
+                        if (input.chain !== chain) throw new Error('conflicting chain');
+                    } else {
+                        chain = input.chain;
+                    }
+                }
```

Comment:
We can use guard clauses a follow a 'happy path', generally nesting lost of if logic is hard to read :) 
 ( double check that i wrote this out correctly) 
```suggestion
                 if (input.account !== this.#account) throw new Error('invalid account'); 
                 
                 if (!input.chain) continue;
                 
                 if (!isSolanaChain(input.chain)) throw new Error('invalid chain');
                 
                 if (chain && input.chain !== chain) throw new Error('conflicting chain');
                 
                 chain = input.chain; 
```